src/stactools/nrcan_landcover/stac.py

```diff
@@ -176,5 +261,88 @@ def create_collection(metadata: Dict[str, Any],
+            href=thumbnail_url,
+            media_type=pystac.MediaType.JPEG,
+            roles=["thumbnail"],
+            title="Land cover of Canada thumbnail",
+        ),
+    )
+
+    collection_label = LabelExtension.summaries(collection,
+                                                add_if_missing=True)
+    collection_label.label_type = [LabelType.RASTER]
+    collection_label.label_tasks = [LabelTask.CLASSIFICATION]
+    collection_label.label_properties = None
+    collection_label.label_classes = [
+        # TODO: The STAC Label extension JSON Schema is incorrect.
+        # https://github.com/stac-extensions/label/pull/8
+        # https://github.com/stac-utils/pystac/issues/611
+        # When it is fixed, this should be None, not the empty string.
+        LabelClasses.create(list(CLASSIFICATION_VALUES.values()), "")
+    ]
+
+    collection_proj = ProjectionExtension.summaries(collection,
+                                                    add_if_missing=True)
+    collection_proj.epsg = [LANDCOVER_EPSG]
+
+    collection_item_asset = ItemAssetsExtension.ext(collection,
+                                                    add_if_missing=True)
+    collection_item_asset.item_assets = {
+        "metadata":
+        AssetDefinition(
+            dict(
+                type=pystac.MediaType.JSON,
+                roles=["metadata"],
+                title="Land cover of Canada metadata",
+            )),
+        "thumbnail":
+        AssetDefinition(
+            dict(
+                type=pystac.MediaType.JPEG,
+                roles=["thumbnail"],
+                title="Land cover of Canada thumbnail",
+            )),
+        "landcover":
+        AssetDefinition({
+            "type":
+            pystac.MediaType.COG,
+            "roles": [
+                "data",
+                "labels",
+                "labels-raster",
+            ],
+            "title":
+            "Land cover of Canada COG",
+            "raster:bands": [
+                RasterBand.create(nodata=0,
+                                  sampling=Sampling.AREA,
+                                  data_type=DataType.UINT8,
+                                  spatial_resolution=30).to_dict()
+            ],
+            "file:values": [{
+                "values": [value],
+                "summary": summary
+            } for value, summary in CLASSIFICATION_VALUES.items()],
+            "label:type":
+            collection_label.label_type[0],
+            "label:tasks":
+            collection_label.label_tasks,
+            "label:properties":
+            None,
+            "label:classes": [collection_label.label_classes[0].to_dict()],
+            "proj:epsg":
+            collection_proj.epsg[0]
+        }),
+        "extent":
+        AssetDefinition(
+            dict(
+                type=pystac.MediaType.GEOJSON,
+                roles=["extent"],
+                title="Land cover of Canada extent",
+            )),
+    }
```

Comment:
All these asset definitions can be reused when making the actual item assets (i.e. `ASSET_DEFINITION.create_asset(href)`). Not a dealbreaker, but could split those out into an assets.py (or add to constants.py) and import.

Reply:
I think there's an issue with the difference between `Asset` and `AssetDefinition`. But it would be nice to separate this out.